src/model/guild/member.rs

```diff
@@ -557,24 +498,8 @@ impl Member {
         &mut self,
         http: impl AsRef<Http>,
         role_ids: &[RoleId],
-    ) -> Result<Vec<RoleId>> {
-        self.roles.retain(|r| !role_ids.contains(r));
-
-        let mut builder = EditMember::default();
-        builder.roles(&self.roles);
-
-        match http
-            .as_ref()
-            .edit_member(self.guild_id.get(), self.user.id.get(), &builder, None)
-            .await
-        {
-            Ok(member) => Ok(member.roles),
-            Err(why) => {
-                self.roles.extend_from_slice(role_ids);
-
-                Err(why)
-            },
-        }
+    ) -> Result<()> {
+        self.edit(http, |b| b.roles(role_ids)).await
```

Comment:
This has the same problem as `add_roles`. You will have to do the opposite: filter out member's roles that are in `role_ids`:
```suggestion
        self.edit(http, |b| b.roles(self.roles.iter().filter(|r| !role_ids.contains(r)))).await
```